src/bindings/js/node/src/model_wrap.cpp

```diff
@@ -128,3 +129,27 @@ Napi::Value ModelWrap::is_dynamic(const Napi::CallbackInfo& info) {
     const auto result = _model->is_dynamic();
     return Napi::Boolean::New(env, result);
 }
+
+Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
+    if (info.Length() != 1 || !info[0].IsNumber()) {
+        reportError(info.Env(), "Invalid argument. Expected a single number.");
+        return Napi::Value();
+    }
+
+    auto idx = info[0].As<Napi::Number>().Int32Value();
+    auto cm_outputs = _model->outputs();  // Output<Node>
+
+    if (idx < 0 || idx >= cm_outputs.size()) {
+        reportError(info.Env(), "Invalid index. Index out of range.");
+        return Napi::Value();
+    }
+
+    auto shape = cm_outputs[idx].get_shape();
+    Napi::Array js_shape = Napi::Array::New(info.Env(), shape.size());
+
+    size_t i = 0;
+    for (auto& dim : shape)
+        js_shape[i++] = Napi::Number::New(info.Env(), dim);
+
+    return js_shape;
```

Comment:
To do not convert it manually, you can use this approach:
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/js/node/src/node_output.cpp#L83